example-app/src/main/java/com/adyen/checkout/example/di/StorageModule.kt

```diff
@@ -28,9 +31,19 @@ object StorageModule {
         PreferenceManager.getDefaultSharedPreferences(appContext)
 
     @Provides
-    fun provideKeyValueStorage(appContext: Application, sharedPreferences: SharedPreferences): KeyValueStorage =
-        DefaultKeyValueStorage(appContext, sharedPreferences)
+    fun provideSharedPreferencesManager(sharedPreferences: SharedPreferences): SharedPreferencesManager =
```

Comment:
This class can be simplified to:
```Kotlin
@Module
@InstallIn(SingletonComponent::class)
internal abstract class StorageModule {

    @Binds
    abstract fun bindKeyValueStorage(defaultKeyValueStorage: DefaultKeyValueStorage): KeyValueStorage

    companion object {

        @Provides
        fun provideSharedPreferences(appContext: Application): SharedPreferences =
            PreferenceManager.getDefaultSharedPreferences(appContext)

        @Provides
        fun provideAssetManager(appContext: Application): AssetManager = appContext.assets
    }
}
```
You have to add `@Inject constructor` to `DefaultKeyValueStorage` though